src/server/scripts/Spells/spell_priest.cpp

```diff
@@ -2788,8 +2789,31 @@ class spell_pri_vampiric_touch : public AuraScript
     }
 };
 
+// 193063 - Protective Light (Aura)
+class spell_pri_protective_light : public AuraScript
+{
+    PrepareAuraScript(spell_pri_protective_light);
+
+    bool CheckEffectProc(const AuraEffect* /*aurEff*/, ProcEventInfo& eventInfo)
+    {
+        return eventInfo.GetProcTarget() == GetCaster();
+    }
+
+    void HandleEffectProc(AuraEffect* auraEff, ProcEventInfo& eventInfo)
+    {
+        eventInfo.GetProcTarget()->CastSpell(GetCaster(), SPELL_PRIEST_PROTECTIVE_LIGHT_AURA, auraEff);
```

Comment:
would use `GetCaster()` here as its shorter and the same unit, thats just personal preference
```suggestion
        GetCaster()->CastSpell(GetCaster(), SPELL_PRIEST_PROTECTIVE_LIGHT_AURA, auraEff);
```